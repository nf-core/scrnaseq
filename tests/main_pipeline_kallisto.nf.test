nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    test("test-dataset_kallisto_aligner") {

        when {
            // the rest is taken from shared config
            params {
                aligner    = 'kallisto'
                outdir     = "${outputDir}/results_kallisto"
            }
        }

        then {

            assertAll(

                //
                // General assertions
                //

                // Did it finish successfully?
                {assert workflow.success},

                // How many tasks were executed?
                {assert workflow.trace.tasks().size() == 19},

                // How many results were produced?
                {assert path("${outputDir}/results_kallisto").list().size() == 4},
                {assert path("${outputDir}/results_kallisto/kallisto").list().size() == 3},
                {assert path("${outputDir}/results_kallisto/kallisto/mtx_conversions").list().size() == 8},
                {assert path("${outputDir}/results_kallisto/kallisto/Sample_X.count").list().size() == 12},
                {assert path("${outputDir}/results_kallisto/kallisto/Sample_Y.count").list().size() == 12},
                {assert path("${outputDir}/results_kallisto/fastqc").list().size() == 12},
                {assert path("${outputDir}/results_kallisto/multiqc").list().size() == 3},

                //
                // Check if files were produced
                //
                {assert new File( "${outputDir}/results_kallisto/kallisto/mtx_conversions/Sample_X/Sample_X_raw_matrix.h5ad" ).exists()},
                {assert new File( "${outputDir}/results_kallisto/kallisto/mtx_conversions/Sample_X/Sample_X_filtered_matrix.h5ad" ).exists()},
                {assert new File( "${outputDir}/results_kallisto/kallisto/mtx_conversions/Sample_Y/Sample_Y_raw_matrix.h5ad" ).exists()},
                {assert new File( "${outputDir}/results_kallisto/kallisto/mtx_conversions/Sample_Y/Sample_Y_filtered_matrix.h5ad" ).exists()},

                //
                // Check if files are the same
                //
                {assert snapshot(
                    path( "${outputDir}/results_kallisto/kallisto/Sample_X.count/counts_unfiltered/cells_x_genes.barcodes.txt" ),
                    path( "${outputDir}/results_kallisto/kallisto/Sample_X.count/counts_unfiltered/cells_x_genes.genes.txt"    ),
                    path( "${outputDir}/results_kallisto/kallisto/Sample_X.count/counts_unfiltered/cells_x_genes.mtx"          ),
                    path( "${outputDir}/results_kallisto/kallisto/Sample_X.count/counts_filtered/cells_x_genes.barcodes.txt" ),
                    path( "${outputDir}/results_kallisto/kallisto/Sample_X.count/counts_filtered/cells_x_genes.genes.txt"    ),
                    path( "${outputDir}/results_kallisto/kallisto/Sample_X.count/counts_filtered/cells_x_genes.mtx"          ),

                    path( "${outputDir}/results_kallisto/kallisto/Sample_Y.count/counts_unfiltered/cells_x_genes.barcodes.txt" ),
                    path( "${outputDir}/results_kallisto/kallisto/Sample_Y.count/counts_unfiltered/cells_x_genes.genes.txt"    ),
                    path( "${outputDir}/results_kallisto/kallisto/Sample_Y.count/counts_unfiltered/cells_x_genes.mtx"          ),
                    path( "${outputDir}/results_kallisto/kallisto/Sample_Y.count/counts_filtered/cells_x_genes.barcodes.txt" ),
                    path( "${outputDir}/results_kallisto/kallisto/Sample_Y.count/counts_filtered/cells_x_genes.genes.txt"    ),
                    path( "${outputDir}/results_kallisto/kallisto/Sample_Y.count/counts_filtered/cells_x_genes.mtx"          ),

                    path( "${outputDir}/results_kallisto/kallisto/mtx_conversions/Sample_X/Sample_X_raw_matrix.seurat.rds"     ),
                    path( "${outputDir}/results_kallisto/kallisto/mtx_conversions/Sample_X/Sample_X_raw_matrix.sce.rds"        ),
                    path( "${outputDir}/results_kallisto/kallisto/mtx_conversions/Sample_X/Sample_X_filtered_matrix.seurat.rds"     ),
                    path( "${outputDir}/results_kallisto/kallisto/mtx_conversions/Sample_X/Sample_X_filtered_matrix.sce.rds"        ),

                    path( "${outputDir}/results_kallisto/kallisto/mtx_conversions/Sample_Y/Sample_Y_raw_matrix.seurat.rds"     ),
                    path( "${outputDir}/results_kallisto/kallisto/mtx_conversions/Sample_Y/Sample_Y_raw_matrix.sce.rds"        ),
                    path( "${outputDir}/results_kallisto/kallisto/mtx_conversions/Sample_Y/Sample_Y_filtered_matrix.seurat.rds"     ),
                    path( "${outputDir}/results_kallisto/kallisto/mtx_conversions/Sample_Y/Sample_Y_filtered_matrix.sce.rds"        )
                ).match()}

            ) // end of assertAll()

        }
    }

}
