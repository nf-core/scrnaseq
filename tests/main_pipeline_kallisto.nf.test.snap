{
    "test-dataset_kallisto_aligner": {
        "content": [
            "cells_x_genes.barcodes.txt:md5,72d78bb1c1ee7cb174520b30f695aa48",
            "cells_x_genes.genes.txt:md5,acd9d00120f52031974b2add3e7521b6",
            "cells_x_genes.mtx:md5,894d60da192e3788de11fa8fc1fa711d",
            "cells_x_genes.barcodes.txt:md5,9e456448a0d4da992d4ca89987c1feba",
            "cells_x_genes.genes.txt:md5,acd9d00120f52031974b2add3e7521b6",
            "cells_x_genes.mtx:md5,6cf18c72ed8d4f96e4f01d2bf637f962",
            "cells_x_genes.barcodes.txt:md5,a8cf7ea4b2d075296a94bf066a64b7a4",
            "cells_x_genes.genes.txt:md5,acd9d00120f52031974b2add3e7521b6",
            "cells_x_genes.mtx:md5,abd83de117204d0a77df3c92d00cc025",
            "cells_x_genes.barcodes.txt:md5,32464c83608493daf0a22dc39ede6bc8",
            "cells_x_genes.genes.txt:md5,acd9d00120f52031974b2add3e7521b6",
            "cells_x_genes.mtx:md5,fb032cbac4888032c81ed79a740fa181",
            "Sample_X_raw_matrix.seurat.rds:md5,6dba7ab652441df6a2b0712c7529053b",
            "Sample_X_raw_matrix.sce.rds:md5,3fef29fea599561551a06caa82811e2b",
            "Sample_X_filtered_matrix.seurat.rds:md5,a25a8c6d0b26facffaae352c2f2742fc",
            "Sample_X_filtered_matrix.sce.rds:md5,554e1376664bbff371cb09c39a969d06",
            "Sample_Y_raw_matrix.seurat.rds:md5,a9e9ac3d1bf83f4e791d6f0c3f6540de",
            "Sample_Y_raw_matrix.sce.rds:md5,6818392c6b8b65d762521406aa963b2a",
            "Sample_Y_filtered_matrix.seurat.rds:md5,f872ac3625f2b2f1fca77a5b736cc18f",
            "Sample_Y_filtered_matrix.sce.rds:md5,e93e974427f87e8774636c64d33d628e"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.1"
        },
        "timestamp": "2025-01-17T10:13:54.332011219"
    }
}